File: dashboard-backend/src/main/java/com/quanluu/dashboard/service/TagService.java
package com.quanluu.dashboard.service;

import com.quanluu.dashboard.model.Tag;

import java.util.List;
import java.util.Optional;

public interface TagService {

    List<Tag> getAllTags();

    Optional<Tag> getTagById(Long id);

    Optional<Tag> getTagByName(String name);

    Tag createTag(Tag tag);

    Tag updateTag(Long id, Tag tagDetails);

    void deleteTag(Long id);
}

File: dashboard-frontend/src/services/tagService.js
import axios from 'axios'

const API_URL = 'http://localhost:8080/api/tags'

export default {
  getAllTags() {
    return axios.get(API_URL)
  },

  getTagById(id) {
    return axios.get(`${API_URL}/${id}`)
  },

  createTag(tag) {
    return axios.post(API_URL, tag)
  },

  updateTag(id, tag) {
    return axios.put(`${API_URL}/${id}`, tag)
  },

  deleteTag(id) {
    return axios.delete(`${API_URL}/${id}`)
  }
}
